modules/core/shared/src/main/scala/lucuma/core/math/Angle.scala

```diff
@@ -23,175 +24,15 @@ import scala.math
-   * @group Conversions
-   */
-  def toSignedDoubleDegrees: Double =
-    Angle.signedMicroarcseconds.get(this).toDouble / Angle.µasPerDegree.toDouble
-
-  /**
-   * This angle in signed decimal degrees. Approximate, non-invertible
-   * @group Conversions
-   */
-  def toSignedBigDecimalDegrees: BigDecimal =
-    BigDecimal(Angle.signedMicroarcseconds.get(this)) / Angle.µasPerDegree
-
-  /**
-   * This angle in decimal radian, [0 .. 2π) Approximate, non-invertible
-   * @group Conversions
-   */
-  def toDoubleRadians: Double =
-    toDoubleDegrees.toRadians
-
-  /**
-   * This angle in signed decimal radians, [-π .. π) Approximate, non-invertible
-   * @group Conversions
-   */
-  def toSignedDoubleRadians: Double =
-    toSignedDoubleDegrees.toRadians
-
-  /**
-   * Sum of this angle and `a`. Exact, commutative, invertible.
-   * @group Operations
-   */
-  def +(a: Angle): Angle =
-    Angle.fromMicroarcseconds(toMicroarcseconds + a.toMicroarcseconds)
-
-  /**
-   * Difference of this angle and `a`. Exact, invertible.
-   * @group Operations
-   */
-  def -(a: Angle): Angle =
-    Angle.fromMicroarcseconds(toMicroarcseconds - a.toMicroarcseconds)
-
-  /**
-   * This angle as an Offset.P. Exact, invertible.
-   *
-   * @group Conversions
-   */
-  def p: Offset.P =
-    Offset.P(this)
-
-  /**
-   * This angle as an Offset.Q. Exact, invertible.
-   *
-   * @group Conversions
-   */
-  def q: Offset.Q =
-    Offset.Q(this)
-
-  /**
-   * This angle as an offset in p. Exact, invertible.
-   *
-   * @group Conversions
-   */
-  def offsetInP: Offset =
-    Offset(p, Offset.Q.Zero)
-
-  /**
-   * This angle as an offset in q. Exact, invertible.
-   *
-   * @group Conversions
-   */
-  def offsetInQ: Offset =
-    Offset(Offset.P.Zero, q)
-
-  /**
-   * Trigonometric sine of the angle.
-   */
-  def sin: Double = math.sin(toDoubleRadians)
-
-  /**
-   * Trigonometric cosine of the angle.
-   */
-  def cos: Double = math.cos(toDoubleRadians)
-
-  /**
-   * Trigonometric tangent of the angle.
-   */
-  def tan: Double = math.tan(toDoubleRadians)
-
-  /** String representation of this Angle, for debugging purposes only. */
-  override def toString: String = {
-    val dms = Angle.dms.get(this)
-    s"Angle.fromDMS(${dms.degrees},${dms.arcminutes},${dms.arcseconds},${dms.milliarcseconds},${dms.microarcseconds})"
-  }
-
-  /** Angles are equal if their magnitudes are equal. Exact. */
-  override final def equals(a: Any): Boolean =
-    a match {
-      case a: Angle => a.toMicroarcseconds === toMicroarcseconds
-      case _        => false
-    }
-
-  override final def hashCode: Int =
-    toMicroarcseconds.toInt
-
-  /**
-   * angle difference or separation between two angles in the range [0 .. π]
-   */
-  def difference(a: Angle): Angle =
-    Angle.difference(this, a)
-}
+opaque type Angle = Long
 
 object Angle extends AngleOptics {
+  inline private def apply(w: Long): Angle = w
```

Comment:
Could/should we keep the `0 <= toMicroarcseconds < 360º` sanity check? That is, `Angle` isn't just any `Long`, it has to be microarcseconds and in range.  I think this is still guaranteed by the constructors but it seems like it would be easy to slip in a non-valid constructor at some point.

Reply:
I wasn't sure about that but we could restore it. I'm not sure if we can inline in that case

Reply:
Maybe we should `Refine` it?

Reply:
> Maybe we should Refine it?

That sounds right, I think.  We wrote `Angle` long before we embraced refine.